PR #149 present in upstream master:

Block termination signals for the jitter tasks

To improve reliability of daemon shutdown make sure the foreground
thread is the thread catching SIGINT and friends.

---

diff -aur a/rngd_jitter.c b/rngd_jitter.c
--- a/rngd_jitter.c
+++ b/rngd_jitter.c
@@ -257,6 +257,7 @@
 
 static void *thread_entropy_task(void *data)
 {
+	sigset_t blockset;
 	cpu_set_t cpuset;
 
 	ssize_t ret;
@@ -266,6 +267,12 @@
 	int written;
 	/* STARTUP */
 
+	sigemptyset(&blockset);
+	sigaddset(&blockset, SIGINT);
+	sigaddset(&blockset, SIGTERM);
+	sigaddset(&blockset, SIGALRM);
+	pthread_sigmask(SIG_BLOCK, &blockset, NULL);
+
 	/*
 	 * Set our timeout value
 	 * -1 means adaptive, i.e. sleep for the last
